processor/resourcedetectionprocessor/internal/kubeadm/kubeadm.go

```diff
@@ -0,0 +1,62 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package kubeadm // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal/kubeadm"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/processor"
+	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/metadataproviders/kubeadm"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal/kubeadm/internal/metadata"
+)
+
+const (
+	TypeStr = "kubeadm"
+)
+
+var _ internal.Detector = (*detector)(nil)
+
+type detector struct {
+	provider kubeadm.Provider
+	logger   *zap.Logger
+	ra       *metadata.ResourceAttributesConfig
+	rb       *metadata.ResourceBuilder
+}
+
+func NewDetector(set processor.Settings, dcfg internal.DetectorConfig) (internal.Detector, error) {
+	cfg := dcfg.(Config)
+	if err := cfg.UpdateDefaults(); err != nil {
+		return nil, err
+	}
+
+	k8snodeProvider, err := kubeadm.NewProvider(cfg.configMapName, cfg.configMapNamespace, cfg.APIConfig)
```

Comment:
We should probably rename this variable from `k8snodeProvider` to something more relevant for this detector.